src/transactions/TransactionFrame.h

```diff
@@ -312,17 +277,148 @@ class TransactionFrame : public TransactionFrameBase
     bool isSoroban() const override;
     SorobanResources const& sorobanResources() const override;
 
-    bool
-    consumeRefundableSorobanResources(uint32_t contractEventSizeBytes,
-                                      int64_t rentFee, uint32_t protocolVersion,
-                                      SorobanNetworkConfig const& sorobanConfig,
-                                      Config const& cfg);
-
     static FeePair computeSorobanResourceFee(
         uint32_t protocolVersion, SorobanResources const& txResources,
         uint32_t txSize, uint32_t eventsSize,
         SorobanNetworkConfig const& sorobanConfig, Config const& cfg);
     virtual int64 declaredSorobanResourceFee() const override;
     virtual bool XDRProvidesValidFee() const override;
+
+#ifdef BUILD_TESTS
+    friend class TransactionTestFrame;
+#endif
+};
+
+#ifdef BUILD_TESTS
+class TransactionTestFrame;
+using TransactionTestFramePtr = std::shared_ptr<TransactionTestFrame>;
+
+// The normal TransactionFrame object is immutable, and the caller needs to
+// manage mutable result state via TransactionResultPayloadBase. This class
+// wraps an immutable TransactionFrame object with its associated mutable
+// TransactionResultPayloadBase for testing purposes.
+class TransactionTestFrame : public TransactionFrameBase
```

Comment:
I don't think this class definition should live in TransactionFrame.h  (it can be its own file in transactions/test folder)